src/main/java/com/redislabs/redisai/DataType.java

```diff
@@ -188,10 +139,10 @@ public static DataType baseObjType(Object o) {
     throw new IllegalArgumentException("cannot create Tensors of type " + c.getName());
   }
 
-  private static List<byte[]> toByteArray(Object obj, int[] dimensions, int dim, DataType type) {
+  private static List<byte[]> toByteArray(Object obj, long[] dimensions, long dim, DataType type) {
```

Comment:
why is dim long if we cast it on 145 to int?

Reply:
solved in the new commit